packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -43,9 +53,15 @@ const NewNetwork = decorate([
       onChangePif: (_, value) => ({ bonded }) =>
         bonded ? { pifs: value } : { pif: value },
       reset: () => EMPTY,
-      toggleBonded: () => ({ bonded }) => ({
+      toggleBonded: () => ({ bonded, isPrivate }) => ({
         ...EMPTY,
         bonded: !bonded,
+        isPrivate: bonded ? isPrivate : false,
+      }),
```

Comment:
```js
toggleBonded() {
    const { bonded, isPrivate } = this.state
    return {
        ...EMPTY,
        bonded: !bonded,
        isPrivate: bonded ? isPrivate : false,
    }
}
```

Reply:
Try to skip reducers

Reply:
@HamadaBrest Try to explain the reasoning behind a comment, especially to someone new on the codebase.

@benjamreis We are using a framework called `reaclette` which is based on `freactal`. It's stays compatible with its API but will drop it in a future version, so it's better to use its own API (which is better for async processes):

```js
// instead of
foo: (effects, arg1, arg2) => (state, props) => newState,

// do
foo(effects, arg1, arg2) {
  this.props
  this.state
  return newState
}
```